zips/zip-0233.md

```diff
@@ -19,224 +20,128 @@ The key words "MUST", "SHOULD", "SHOULD NOT", "MAY", "RECOMMENDED", "OPTIONAL",
-
-### Coinbase Transactions
-
-Any excess miner fee/reward is sent to the ZSF.
-
-In new blocks, this deposit is tracked via the `ZSF_DEPOSIT` field in coinbase
-transactions. 
-
-If the `ZSF_DEPOSIT` field is not present in a coinbase transaction with an
-older transaction version, it is defined as the value of any unspent miner fee
-and miner reward. This re-defines these previously unspendable funds as ZSF
-deposits.
-
-#### Consensus Rule Changes
-
--  The remaining value in the transparent transaction value pool of a coinbase
-   transaction is **deposited in the sustainability fund**.
-
-### `ZSF_DEPOSIT` Requirements
-
-- ZIP 230 [3] must be updated to include `ZSF_DEPOSIT`.
-- ZIP 244 [4] must be updated as well to include `ZSF_DEPOSIT`.
+The modifications required are:
+
+1. The addition of a transaction field representing an amount to burn for a
+   given transaction.
+2. The inclusion of the burn amount in the calculated total output value for a
+   given transaction.
+3. Consensus rules to ensure the burned amount is no longer available for
+   circulation on the network.
+
+## Transaction Format
+
+The following field is added to the v6 transaction format [3]:
+
++-------+----------------+------------+------------------------------------------------------+
+| Bytes | Name           | Data Type  | Description                                          |
++=======+================+============+======================================================+
+|   8   | ``burnAmount`` | ``uint64`` | The amount of input value to be burned, in zatoshis. |
++-------+----------------+------------+------------------------------------------------------+
+
+**Note:** Older transaction versions can continue to be supported after a
+network upgrade, but burning is not possible for these transactions. For
+example, NU5 supports both v4 and v5 transaction formats, for both coinbase and
+non-coinbase transactions.
+
+## Consensus Rules
+
+The burned value must now be considered when calculating the total output value
+of a transaction. It should be treated similarly to a transparent output,
+except that there is no way for this value to be used as an input in a future
+transaction.
+
+## Digests
+
+The transaction digest algorithm defined in ZIP 244 [4] is to be modified for v6
+transactions as follows:
+
+Section T.1: header_digest [5] is specified in draft-txv6-sighash  [6] to read:
+
+> A BLAKE2b-256 hash of the following values:
+>
+> ```
+> T.1a: version             (4-byte little-endian version identifier including ``fOverwintered`` flag)
+> T.1b: version_group_id    (4-byte little-endian version group identifier)
+> T.1c: consensus_branch_id (4-byte little-endian consensus branch id)
+> T.1d: lock_time           (4-byte little-endian ``nLockTime`` value)
+> T.1e: expiry_height       (4-byte little-endian block height)
+> T.1f: burn_amount         (8-byte little-endian burn amount value)
```

Comment:
This has an interaction with adding the explicit fee field, but the intent is clear.